project-backend/Controllers/JobController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using project_backend.Models;
+using project_backend.Models.Job;
+using project_backend.Models.JobController.GetJobs;
+using project_backend.Providers.JobProvider;
+using System;
+using System.Linq;
+
+namespace project_backend.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class JobController : ControllerBase
+    {
+        private readonly ILogger<JobController> _logger;
+        private readonly IJobProvider _jobProvider;
+
+        public JobController(ILogger<JobController> logger, IJobProvider jobProvider)
+        {
+            _logger = logger;
+            _jobProvider = jobProvider;
+        }
+
+        [HttpPost]
+        [Authorize]
+        [ProducesResponseType(typeof(GetJobsResponseObject), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(Error), StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+        public IActionResult GetJobs([FromBody] GetJobsQueryObject query)
+        {
+            _logger.LogInformation("GetJobs({0})", query);
+
+            if (!int.TryParse(User.FindFirst("Id").Value, out var userId))
+            {
+                return Unauthorized(new Error("Not logged in"));
+            }
+
+            var jobs = _jobProvider.QueryJobs();
+            if (query.ByCurrentUserOnly ?? false)
+            {
+                jobs = jobs.Where(j => j.UserId == userId);
+            }
+
+            if (query.OlderThan != null)
+            {
+                jobs = jobs.Where(j => j.PostDate <= query.OlderThan);
```

Comment:
This method is quite large and should be split in some smaller methods

Reply:
Fixed this in https://github.com/PC-Chair-Fight/project-backend/pull/17/commits/9bdf23967d4bc518ac157d157f3942ac417c833f. Please review the changes again.